cuda_thrust_sort/src/openmc_thrust_sort.cu

```diff
@@ -1,18 +1,23 @@
-#include<thrust/sort.h>
+#include <thrust/sort.h>
 
 // This macro is used to enable the "__host__ __device__" attributes
 // for the EventQueueItem comparator in event.h. If those attributes
 // are missing, the code will compile and run, but it will use some sort
 // of default comparator that does not have the desired effect.
 #define COMPILE_CUDA_COMPARATOR
 
-#include"openmc/event.h"
+#include "openmc/event.h"
 
 namespace openmc{
 
-void device_sort_event_queue_item(EventQueueItem* begin, EventQueueItem* end)
+void thrust_sort_MatE(EventQueueItem* begin, EventQueueItem* end)
 {
-  thrust::sort(thrust::device, begin, end);
+  thrust::sort(thrust::device, begin, end, MatECmp());
+}
+
+void thrust_sort_CellSurf(EventQueueItem* begin, EventQueueItem* end)
+{
+  thrust::sort(thrust::device, begin, end, CellSurfCmp());
```

Comment:
As we discussed offline, you could probably have this be a template function with the functor being the template parameter and then use explicit template instantiation. However, with only two versions, it's not too bad so we can also just forgo that for now. If you decide to leave as is, can you just put a TODO above this noting that it could be done as a template function?

Reply:
Sounds good -- I've added the TODO tags for now, and will loop back on this down the line